pytorch_lightning/profiler/base.py

```diff
@@ -25,7 +25,14 @@
 
 
 class AbstractProfiler(ABC):
-    """Specification of a profiler."""
+    """Specification of a profiler.
+
+    See deprecation message below
+
+    .. deprecated:: v1.6
```

Comment:
```suggestion
    """Specification of a profiler.

    .. deprecated:: v1.6
```

Reply:
That line is actually a workaround for a bug in docformatter https://github.com/PyCQA/docformatter/issues/78.
I've seen this line show up in some other places as well:
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/loggers/base.py#L187.

Reply:
Would this also work as a workaround?
```
"""
Specification of a profiler.

    .. deprecated:: v1.6
"""
```

Reply:
Unfortunately that doesn't work. Docformatter moves `Specification of a profiler` up one line, and then the same bug returns

Reply:
```
""""Specification of a profiler.

.. deprecated:: v1.6
    :class:`AbstractProfiler` was ...
```
can you try this? if this doesn't work, keep it just like it is now.

or if you end up removing the class altogether, then it won't be a problem anyways.

Reply:
hmm, I'm just wondering how I was able to do this in https://github.com/PyTorchLightning/pytorch-lightning/pull/12084/files#diff-667a2513c158c2b01d73b479e4dea75587d8531f9cd286891d34570a2fd145dcR2216

Reply:
> hmm, I'm just wondering how I was able to do this in https://github.com/PyTorchLightning/pytorch-lightning/pull/12084/files#diff-667a2513c158c2b01d73b479e4dea75587d8531f9cd286891d34570a2fd145dcR2216

Are you referring to line 2217? I think that works because there is a separate line with text above the deprecation message. This bug only appears if there is no text between the top line of text and the deprecation warning.

Reply:
You can see the same issue appears over here:
Commit: https://github.com/PyTorchLightning/pytorch-lightning/pull/11832/commits/27913e4b8002dfc91cb8c1a527a8e8822ef2b4c9
Commit after docformatter: https://github.com/PyTorchLightning/pytorch-lightning/pull/11832/commits/1b7a1bd05ac2fdecc7c1d765c54e3056b23a28b3
The docformatter bug appears in `_reduce_agg_metrics()` and `_finalize_agg_metrics()`, but notably it does not affect `_aggregate_metrics()`. It seems like this bug is not an issue if there is a new line with text either above or below the deprecation message (not include the top line).

Reply:
> ```
> """"Specification of a profiler.
> 
> .. deprecated:: v1.6
>     :class:`AbstractProfiler` was ...
> ```
> 
> can you try this? if this doesn't work, keep it just like it is now.
> 
> or if you end up removing the class altogether, then it won't be a problem anyways.

I gave this a try and it ended up formatting to
```
""""Specification of a profiler.

.. deprecated:: v1.6   :class:`AbstractProfiler` was ...
```
This is the exact same issue as here:
https://github.com/PyTorchLightning/pytorch-lightning/commit/1b7a1bd05ac2fdecc7c1d765c54e3056b23a28b3
If we can find a better workaround then there are 4 other places we should fix too.